pkg/cli/interface.go

```diff
@@ -0,0 +1,186 @@
+}
+
+// Domain contains information regarding a domain(s).
+// As different methods are called on it, the fields within the structure are updated accordingly.
+// Call its appropriate method to get/list/update a Domain.
+type Domain struct {
+	ID     string
+	Name   string
+	Opts   Options
+	Result domains.CommonResult
+	IsList bool
+}
+
+// Project contains information regarding a project(s).
+// As different methods are called on it, the fields within the structure are updated accordingly.
+// Call its appropriate method to get/list/update a Project.
+type Project struct {
+	ID         string
+	Name       string
+	DomainID   string
+	DomainName string
+	Opts       Options
+	Result     projects.CommonResult
+	IsList     bool
+}
+
+// Options contains different options that affect the output of a get/list/update operation.
+type Options struct {
+	Names    bool
+	Long     bool
+	Area     string
+	Service  string
+	Resource string
+}
+
+// GetTask is the interface type that abstracts a get operation.
+type GetTask interface {
+	get()
+	getJSON() interface{}
+	renderCSV() *csvData
+}
+
+// RunGetTask is the function that operates on a GetTask and shows the output in the respective
+// format that is specified at the command line.
+func RunGetTask(t GetTask, outputFmt string) {
+	t.get()
+	switch outputFmt {
+	case "json":
+		writeJSON(t.getJSON())
+	case "csv":
+		t.renderCSV().writeCSV()
+	default:
+		t.renderCSV().writeTable()
+	}
+}
+
+// ListTask is the interface type that abstracts a list operation.
+type ListTask interface {
+	list()
+	getJSON() interface{}
+	renderCSV() *csvData
+}
+
+// RunListTask is the function that operates on a ListTask and shows the output in the respective
+// format that is specified at the command line.
+func RunListTask(t ListTask, outputFmt string) {
+	t.list()
+	switch outputFmt {
+	case "json":
+		writeJSON(t.getJSON())
+	case "csv":
+		t.renderCSV().writeCSV()
+	default:
+		t.renderCSV().writeTable()
+	}
+}
+
+// Resource contains quota information about a single resource.
+type Resource struct {
+	Name    string
+	Value   int64
+	Unit    limes.Unit
+	Comment string
+}
+
+// Quotas is a map of service name to a list of resources. It contains the aggregate
+// quota values used by the set methods to update a single cluster/domain/project.
+type Quotas map[string][]Resource
+
+// SetTask is the interface type that abstracts a put operation.
+type SetTask interface {
+	set(*Quotas)
+}
+
+// RunSetTask is the function that operates on a SetTask and shows the output in the respective
+// format that is specified at the command line.
+func RunSetTask(t SetTask, q *Quotas) {
+	t.set(q)
+}
+
+// RunSyncTask schedules a sync job that pulls quota and usage data for a project from
+// the backing services into Limes' local database.
+func RunSyncTask(p *Project) {
+	_, limesV1 := getServiceClients()
+
+	err := projects.Sync(limesV1, p.DomainID, p.ID, projects.SyncOpts{})
+	handleError("could not sync project: %v", err)
```

Comment:
`%v` is not recognized by `handleError`. I guess that's just a leftover from a previous iteration?